compiler/hash-types/src/new/control.rs

```diff
@@ -53,3 +53,30 @@ pub enum LoopControlTerm {
     Break,
     Continue,
 }
+
+/// A conditional pattern, containing a pattern and an condition.
+///
+/// This is `A if c` in its most general form, where `A` is a pattern and `c` is
+/// a boolean-valued term.
+///
+/// This pattern matches iff `A` matches and `c` is met.
+#[derive(Clone, Debug, Copy)]
+pub struct IfPat {
+    /// The subject pattern `A`.
+    pub pat: PatId,
+    /// The condition `c`.
+    pub condition: TermId,
```

Comment:
If the condition is always a boolean type, is there any need to store the term here?

Reply:
The condition is of a boolean type, but it can still be any arbitrary term that is typed as a boolean